### PR TITLE
remade get_random_default_weapon() to decrement enemy's hp, strength, or defense

### PR DESCRIPTION
**Pull request by sonnykaiyuenlee**
updated battle_logic to ensure the functions are fed the correct combatant (player or enemy) depending on whether the item is a consumable or a weapon. Also completely changed get_random_default_weapon() in battle_default_objects to have negative values which streamlined the process of consuming/using the default objects module/weapons in general.

### CODE REVIEW

File: include/battle/battle_logic.h [hunk 1]
```diff
@@ -59,15 +59,16 @@ battle_item_t *find_battle_item(battle_item_t *inventory, int id);
...
  *
  *  Parameters: 
  *   c - combatant information
+ *   battle - battle information   
  *   id - the number id of the battle_item
  *  Returns:
  *   SUCCESS or FAILURE
  */
-int use_battle_item(combatant_t *c,int id);
+int use_battle_item(combatant_t *c, battle_t *battle, int id);
```
Comment (lczielinski) [inline include/battle/battle_logic.h:71]: This change needs to be updated in battle_command_example.c when this function is called

File: src/battle/src/battle_default_objects.c [hunk 2]
```diff
@@ -22,26 +22,35 @@ battle_item_t *get_random_default_weapon()
...
-    char* name_array[]= {"DAGGER", "DUAL KNIVES", "CROSSBOW", "GUT BLASTER",  "LASER GUN", 
-                         "SAW DISK", "BRASS KNUCKLES", "KATANA", "TASER", "BAZOOKA"};
-
+    int rand = randnum(1, 6); 
+    char* name_array[]= {"SWORD", "HAMMER", "SLIME", "SLEEPING GAS", "SQUID INK", "LAUGHING GAS"};
+    char* description_array[] = {"Reduces enemy's HP by 20", "Reduces enemy's HP by 10", 
+                                 "Reduces enemy's ATTACK by 5", "Reduces enemy's ATTACK by 10",
+                                 "Reduces enemy's DEFENSE by 10", "Reduces enemy's DEFENSE by 15"};
+    int hp_array[] = {-20, -10, 0, 0, 0, 0};
+    int attack_array[] = {-30, -20, 0, 0, 0, 0};
```
Comment (lczielinski) [inline src/battle/src/battle_default_objects.c:31]: Shouldn't the attack_array correspond to the 3rd and 4th weapons? If so, would it look more like: {0, 0, -5, -10, 0, 0}

File: src/battle/src/battle_default_objects.c [hunk 3]
```diff
@@ -22,26 +22,35 @@ battle_item_t *get_random_default_weapon()
...
+    int defense_array[] = {0, 0, -2, -5, -10, -15};
+    int durability_array[] = {100, 80, 60, 40, 30, 20};
 
     rv_weapon->id = rand;
     rv_weapon->is_weapon = true;
-    rv_weapon->effectiveness_decrement = rand * 2;
-    rv_weapon->quantity = 1;
-    rv_weapon->durability = rand * 10; 
+    //set effectiveness_decrement?
```
Comment (lczielinski) [inline src/battle/src/battle_default_objects.c:37]: I think set this to a default 10 for now?

File: src/battle/src/battle_default_objects.c [hunk 4]
```diff
@@ -22,26 +22,35 @@ battle_item_t *get_random_default_weapon()
     battle_item_t *rv_weapon = calloc(1, sizeof(battle_item_t));
     assert(rv_weapon != NULL);
 
-    int rand = randnum(1,10); 
-    char* name_array[]= {"DAGGER", "DUAL KNIVES", "CROSSBOW", "GUT BLASTER",  "LASER GUN", 
-                         "SAW DISK", "BRASS KNUCKLES", "KATANA", "TASER", "BAZOOKA"};
-
+    int rand = randnum(1, 6); 
+    char* name_array[]= {"SWORD", "HAMMER", "SLIME", "SLEEPING GAS", "SQUID INK", "LAUGHING GAS"};
```
Comment (lczielinski) [inline src/battle/src/battle_default_objects.c:26]: This is only if you have time, but maybe change the names of the items to just have the first letter capitalized to match the others

**Review by lczielinski — CHANGES_REQUESTED** [review]
Looks good overall, I think the biggest change is that you need to change the function calls to use_battle_item() in battle_command_example.c to match the new ones, and I made a few other comments too. Very creative weapon names and I can't wait for this to be integrated in the example!

**Review by lczielinski — APPROVED** [review]
Looks really great!!

File: include/battle/battle_default_objects.h [hunk 5]
```diff
@@ -22,22 +22,10 @@
...
- * Note: Currently, our consume_item function works with consumables, not weapons.
- * We will leave this function for the future if using weapons is implemented.
- */
-battle_item_t *get_random_default_weapon();
-
 /*
  * Generates a random consumable item from a list of previously-initialized consumables
  *
- * Parameters: none
+ * Parameters: nonest
```
Comment (firatciftci) [inline include/battle/battle_default_objects.h:28]: This should probably be reverted back to `none`.

File: src/battle/src/battle_logic.c [hunk 6]
```diff
@@ -89,8 +89,15 @@ int use_battle_item(combatant_t *c, int id)
         return FAILURE;
     }
 
-    consume_battle_item(c, item);
-    item->quantity -= 1;
+    if(item->is_weapon)
```
Comment (firatciftci) [inline src/battle/src/battle_logic.c:92]: Make sure to leave a blank space between `if` and `(` on this line.

File: src/battle/src/battle_logic.c [hunk 7]
```diff
@@ -89,8 +89,15 @@ int use_battle_item(combatant_t *c, int id)
...
+    {
+        consume_battle_item(battle->enemy, item);
+        item->durability -= 10;
+    }else
```
Comment (firatciftci) [inline src/battle/src/battle_logic.c:96]: Make sure to leave a blank space between `}` and `else` on this line.

File: tests/battle/test_battle_logic.c [hunk 8]
```diff
@@ -338,6 +341,39 @@ Test(battle_logic, do_not_find_item)
...
+
+    combatant_t *player = combatant_new("player", true, NULL, player_stats, NULL, weapon, BATTLE_AI_NONE);
+
+    combatant_t *enemy = combatant_new("enemy", false, NULL, enemy_stats, NULL, NULL, BATTLE_AI_NONE);
+
+    battle_t *battle = calloc(1, sizeof(battle_t));
+    battle->player = player;
+    battle->enemy = enemy;
+    //battle->environment = NULL;
+    //battle->turn = NULL;
```
Comment (firatciftci) [inline tests/battle/test_battle_logic.c:362]: You should get rid of any unused commented out code.

File: tests/battle/test_battle_logic.c [hunk 9]
```diff
@@ -338,6 +341,39 @@ Test(battle_logic, do_not_find_item)
...
+    int expected_strength = battle->enemy->stats->strength + weapon->attack;
+    int expected_defense = battle->enemy->stats->defense + weapon->defense; 
+    int expected_durability = weapon->durability - 10;
+    use_battle_item(player, battle, weapon->id);
+    cr_assert_eq(battle->enemy->stats->hp, expected_hp, "consume_battle_weapon() does correctly set enemy hp after use");
+    cr_assert_eq(battle->enemy->stats->strength, expected_strength, "consume_battle_weapon() does correctly set enemy strength after use");
+    cr_assert_eq(battle->enemy->stats->defense, expected_defense, "consume_battle_weapon() does correctly set enemy defense after use");
+    cr_assert_eq(player->items->durability, expected_durability, "consume_battle_weapon() does correctly set item durablity after use");
+
+
```
Comment (firatciftci) [inline tests/battle/test_battle_logic.c:374]: No need to have two extra blank lines here.

File: tests/battle/test_battle_logic.c [hunk 10]
```diff
@@ -405,7 +441,10 @@ Test(battle_logic, uses_battle_item_correctly)
     combatant_t *p = combatant_new("Player", true, NULL, pstats, NULL, head, BATTLE_AI_NONE);
     cr_assert_not_null(p, "combatant_new() failed");
 
-    int res = use_battle_item(p, 100);
+    battle_t *battle = calloc(1, sizeof(battle_t));
+    battle->player = p;
+    //battle_t *battle = set_battle(p, NULL, NULL);
```
Comment (firatciftci) [inline tests/battle/test_battle_logic.c:446]: You should get rid of any unused commented out code.

**Review by firatciftci — CHANGES_REQUESTED** [review]
I've requested some minor stylistic fixes.

**Review by firatciftci — APPROVED** [review]
Everything looks good to me!

**Comment by firatciftci** [comment]
Issue Score: ✔️+

Comments:
Good work overall! I only have one thing to comment on:
- When an PR is done, you should include a final comment summarizing what was merged to `dev` and what work remains to be done, if any.

**Pull request merged**